### PR TITLE
Implement HTTP header case adjustment API

### PR DESCRIPTION
**Pull request by Miciah**
* `pkg/operator/controller/ingress/deployment.go` (`RouterHTTPHeaderNameCaseAdjustments`): New constant for the related router environment variable.
(`desiredRouterDeployment`): Set the `ROUTER_H1_CASE_ADJUST` environment variable as appropriate.
* `pkg/operator/controller/ingress/deployment_test.go` (`TestDesiredRouterDeployment`): Verify that `spec.httpHeaders.headerNameCaseAdjustments` has the expected effect.
* `vendor/github.com/openshift/api/operator/v1/*`: Bump.

---

Depends on the following PRs:

- [x] https://github.com/openshift/api/pull/793
- [x] https://github.com/openshift/router/pull/224.

### CODE REVIEW

File: pkg/operator/controller/ingress/deployment_test.go [hunk 1]
```diff
@@ -319,6 +325,11 @@ func TestDesiredRouterDeployment(t *testing.T) {
 
 	checkDeploymentHasEnvVar(t, deployment, "ROUTER_USE_PROXY_PROTOCOL", true, "true")
 
+	checkDeploymentHasEnvVar(t, deployment, "ROUTER_UNIQUE_ID_HEADER_NAME", true, "unique-id")
```
Comment (sgreene570) [inline pkg/operator/controller/ingress/deployment_test.go:328]: The env checks for `ROUTER_UNIQUE_ID_HEADER_NAME` and `ROUTER_UNIQUE_ID_FORMAT` were missing, and this PR just adds them independent of the header case changes, right?

File: pkg/operator/controller/ingress/deployment.go [hunk 2]
```diff
@@ -604,6 +606,15 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		)
 	}
 
+	if ci.Spec.HTTPHeaders != nil && len(ci.Spec.HTTPHeaders.HeaderNameCaseAdjustments) > 0 {
+		var adjustments []string
+		for _, v := range ci.Spec.HTTPHeaders.HeaderNameCaseAdjustments {
+			adjustments = append(adjustments, string(v))
+		}
+		v := strings.Join(adjustments, ",")
```
Comment (sgreene570) [inline pkg/operator/controller/ingress/deployment.go:614]: If `Spec.HTTPHeaders.HeaderNameCaseAdjustments` has a length of `1`, the extra comma after the single entry won't break anything, right?

Comment (sgreene570) [reply pkg/operator/controller/ingress/deployment.go:614]: Ah, never mind, strings.Join handles that case :smile:

File: pkg/operator/controller/ingress/deployment_test.go (hunk 1 above)
Comment (Miciah) [reply pkg/operator/controller/ingress/deployment_test.go:328]: Right, I snuck that in (sorry!).

**Comment by Miciah** [comment]
Rebased.

File: pkg/operator/controller/ingress/deployment_test.go (hunk 1 above)
Comment (sgreene570) [reply pkg/operator/controller/ingress/deployment_test.go:335]: No worries! Just wanted to make sure that was the case.

**Comment by Miciah** [comment]
/retitle Implement HTTP header case adjustment API
now that https://github.com/openshift/router/pull/224 has merged.  The CI test that this PR adds should now pass; I'll keep an eye on e2e-aws-operator...

**Comment by Miciah** [comment]
`TestUniqueIdHeader` and `TestHTTPHeaderCapture` failed.
/test e2e-aws-operator

**Comment by Miciah** [comment]
```
=== RUN   TestHTTPCookieCapture
    operator_test.go:1603: failed to observe the expected log message: timed out waiting for the condition
    operator_test.go:1561: failed to delete pod openshift-ingress/cookietest: pods "cookietest" not found
    operator_test.go:2007: deleted ingresscontroller cookiecapture
--- FAIL: TestHTTPCookieCapture (93.33s)
```
/test e2e-aws-operator

**Pull request merged**